app/factory/Layer.js

```diff
@@ -523,6 +522,40 @@ Ext.define('CpsiMapview.factory.Layer', {
         };
 
         return olSourceProps;
+    },
+
+    /**
+     * Loads and parses the given SLD (by URL) and applies it to the given
+     * vector layer.
+     *
+     * @param  {ol.layer.Vector} mapLayer The layer to apply the style to
+     * @param  {String} sldUrl   The URL to the SLD
+     */
+    loadSld: function (mapLayer, sldUrl) {
+        Ext.Ajax.request({
+            url: sldUrl,
+            method: 'GET',
+            success: function(response) {
+                var sldXml = response.responseText;
+                var sldParser = new GeoStylerSLDParser.SldStyleParser();
+                var olParser = new GeoStylerOpenlayersParser.OlStyleParser(ol);
+
+                sldParser.readStyle(sldXml)
+                    .then(function (gs) {
```

Comment:
Do we need a Promise polyfill?

Reply:
In case this doesn't work as expected, we might e.g. sinply use this [polyfill](https://github.com/getify/native-promise-only/blob/master/README.md) or one of the alternatives.

Reply:
Yes, we should be aware that there might be a need for polyfills, which should be included in the `app.json`, e.g. for Promise and History (see #51) .